src/cmd-buildextend-metal

```diff
@@ -31,6 +31,32 @@ getconfig_def() {
     jq -re .\""$k"\"//\""${default}"\" < "${config}"
 }
 
+# Store information about the artifact into meta.json and also
+# "finalize" it, copying it into the builddir.
+postprocess_artifact() {
+    local artifact_name=$1
+    local local_filepath=$2
+    local target_filename=$3
+    local skip_compress=$4
+    if [[ ! "${skip_compress}" =~ ^(True|False)$ ]]; then
+        fatal "Must specify 'True' or 'False' for skip_compress. Provided: '${skip_compress}'"
+    fi
+    cosa meta --workdir "${workdir}" --build "${build}" --dump | python3 -c "
+import sys, json
+j = json.load(sys.stdin)
+j['images']['${artifact_name}'] = {
+    'path': '${target_filename}',
+    'sha256': '$(sha256sum_str < "${local_filepath}")',
+    'size': $(stat -c '%s' "${local_filepath}"),
+    'skip-compression': ${skip_compress}
```

Comment:
Minor: this is mostly cosmetic, but let's only add the key if it's `true` to retain the current behaviour? That makes it easier to tell when scanning the `meta.json` what artifacts have it.